spring-xd-dirt/src/main/java/org/springframework/xd/dirt/job/dsl/JobSpecification.java

```diff
@@ -0,0 +1,557 @@
+		}
+
+		public String getXmlString() {
+			return xmlString;
+		}
+
+		@Override
+		public void preJobSpecWalk() {
+			try {
+				DocumentBuilderFactory dbf = DocumentBuilderFactory.newInstance();
+				DocumentBuilder db = dbf.newDocumentBuilder();
+				DOMImplementation domImplementation = db.getDOMImplementation();
+
+				// Generate:
+				// <beans xmlns="http://www.springframework.org/schema/beans"
+				//		  xmlns:batch="http://www.springframework.org/schema/batch"
+				//		  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+				//		  xsi:schemaLocation="http://www.springframework.org/schema/beans http://www.springframework.org/schema/beans/spring-beans.xsd
+				//		    http://www.springframework.org/schema/batch http://www.springframework.org/schema/batch/spring-batch.xsd">
+				this.doc = domImplementation.createDocument("http://www.springframework.org/schema/beans", "beans",
+						null);
+				doc.createElementNS("http://www.springframework.org/schema/batch", "batch");
+				doc.getDocumentElement().setAttributeNS("http://www.w3.org/2000/xmlns/", "xmlns:batch",
+						"http://www.springframework.org/schema/batch");
+				doc.getDocumentElement().setAttributeNS("http://www.w3.org/2000/xmlns/", "xmlns:xsi",
+						"http://www.w3.org/2001/XMLSchema-instance");
+				doc.getDocumentElement().setAttributeNS("http://www.w3.org/2000/xsi/", "xsi:schemaLocation",
+						"http://www.springframework.org/schema/beans http://www.springframework.org/schema/beans/spring-beans.xsd");
+				// Setting this 'again' to get it on the front and look more like the above.
+				doc.getDocumentElement().setAttributeNS("http://www.w3.org/2000/xmlns/", "xmlns",
+						"http://www.springframework.org/schema/beans");
+
+				// Generate: <bean id="taskExecutor" class="org.springframework.core.task.SimpleAsyncTaskExecutor"/>
+				Element taskExecutor = doc.createElement("bean");
+				doc.getDocumentElement().appendChild(taskExecutor);
+				taskExecutor.setAttribute("id", "taskExecutor");
+				taskExecutor.setAttribute("class", "org.springframework.core.task.SimpleAsyncTaskExecutor");
+
+				// Generate: <batch:job id="streamName" xmlns="http://www.springframework.org/schema/batch">
+				this.batchJobElement = doc.createElement("batch:job");
+				doc.getDocumentElement().appendChild(batchJobElement);
+				batchJobElement.setAttributeNS("http://www.w3.org/2000/xmlns/", "xmlns",
+						"http://www.springframework.org/schema/batch");
+				if (batchJobId != null) {
+					batchJobElement.setAttribute("id", batchJobId);
+				}
+				this.currentElement.push(batchJobElement);
+			}
+			catch (Exception e) {
+				// todo!
+			}
+		};
+
+		@Override
+		public void postJobSpecWalk() {
+			for (String jobRunnerId : jobRunnerIds) {
+				// Producing:
+				// <bean id="jobRunner-a6e0" class="JobRunningTasklet" scope="step"/>
+				Element bean = doc.createElement("bean");
+				bean.setAttribute("scope", "step");
+				bean.setAttribute("class", "JobRunningTasklet");
```

Comment:
Just a quick update.  The name now is: org.springframework.xd.dirt.batch.tasklet.JobLaunchingTasklet